src/lib/components/Modals/LoginModal.tsx

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import { useForm, DeepMap, FieldError } from 'react-hook-form';
+import {
+  Button,
+  TextField,
+  Grid,
+  Typography,
+  Link,
+  FormControlLabel,
+  Checkbox,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+} from '@material-ui/core';
+import { RegistrationModal } from './RegistrationModal';
+
+export interface IInputs {
+  email: string;
+  password: string;
+}
+
+export const LoginModal: React.FC = () => {
+  const [loginOpen, setLoginOpen] = React.useState(false);
+  const [registrationOpen, setRegistrationOpen] = React.useState(false);
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, handleSubmit, errors } = useForm<IInputs>();
+
+  const handleLoginOpen = () => {
+    setLoginOpen(true);
+  };
+
+  const handleLoginClose = () => {
+    setLoginOpen(false);
+  };
+
+  const handleRegistrationOpen = (
+    event: React.MouseEvent<HTMLSpanElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setRegistrationOpen(true);
+  };
+
+  const handleRegistrationClose = () => {
+    setRegistrationOpen(false);
+    setLoginOpen(false);
+  };
+
+  const onSubmit = (data) => {
+    console.log(data);
+    console.log(errors);
+  };
+
+  const showErrorMessage = (
+    errorsObj: DeepMap<IInputs, FieldError>,
+    inputName: string,
+  ): JSX.Element => {
+    return (
+      errorsObj[inputName] && (
+        // eslint-disable-next-line
+        <span style={{ color: 'red' }}>{errorsObj[inputName].message}</span>
+      )
+    );
+  };
+
+  return (
+    <>
+      <Button variant="outlined" color="primary" onClick={handleLoginOpen}>
+        Log in/Register
+      </Button>
+
+      <Dialog
+        open={loginOpen}
+        onClose={handleLoginClose}
+        aria-labelledby="form-dialog-title"
+      >
+        <DialogTitle id="form-dialog-title">
+          Введіть Ваші email та пароль
+          {/* {response.error && (
+                  <span style={{ color: 'red' }}>Неправильний email або пароль</span>
+                )} */}
+        </DialogTitle>
+        <DialogContent>
+          <form
+            onSubmit={handleSubmit(onSubmit)}
+            style={{ width: '300px', height: '280px', margin: '0 auto' }}
+          >
+            <Grid container justify="center">
+              <Grid item xs={12}>
+                <TextField
+                  name="email"
+                  inputRef={register({
+                    required: {
+                      value: true,
+                      message: "Це поле є обов'язковим",
+                    },
+                    pattern: {
+                      value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
+                      message:
+                        'Email повинен містити символ @ та одну крапку після нього',
+                    },
+                  })}
```

Comment:
I'd suggest you to move these configs out of `return (` statement, that'll be easier to manage them latter

```
{
                    required: {
                      value: true,
                      message: "Це поле є обов'язковим",
                    },
                    pattern: {
                      value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                      message:
                        'Email повинен містити символ @ та одну крапку після нього',
                    },
                  }
```
that pieces that are iniside the register function

Reply:
Should I move it to separate object and then apply to register(), using spread? Like: `register(...emailValidationConfig)`